prometheus_client/metrics.py

```diff
@@ -69,8 +70,24 @@ def _get_use_created() -> bool:
 
 _use_created = _get_use_created()
 
-
 class MetricWrapperBase(Collector):
+    """
+    ``MetricWrapperBase`` is the base class for all metric types.
+    ``MetricWrapperBase`` is not meant to be instantiated directly.
+
+    Each metric types inherits from ``MetricWrapperBase`` must implement
+    ``_metric_init`` and ``_child_samples`` methods.
+
+    Args:
+        name: The name of the metric.
```

Comment:
Could we specify the type as well of the params?

Reply:
Like here: https://realpython.com/python-project-documentation-with-mkdocs/#step-2-create-the-sample-python-package

Reply:
I'm not sure I understand. What it's inferring and where? At least in the code it wouldn't be visible

Reply:
Sorry. https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html

In https://github.com/prometheus/client_python/pull/851 I propse to use napoleon and enable auto typehints. Since the class is already type-hinted, there is no need to also include types in docstring.